Kombit.Samples.Rest/ConfirmationJwtSecurityTokenHandler.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Microsoft.IdentityModel.Tokens;
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Web;
+
+namespace Kombit.Samples.Rest
+{
+    /// <summary>
+    /// This is an example of JwtSecurityTokenHandler to validate the "x5t#S256" confirmation
+    /// </summary>
+    public class ConfirmationJwtSecurityTokenHandler : JwtSecurityTokenHandler
+    {
+        public ClaimsPrincipal ValidateToken(HttpRequestMessage request, TokenValidationParameters validationParameters, out SecurityToken validatedToken)
```

Comment:
This is strange. Shouldn't you override the ValidateToken(token, validationParameters, out validatedToken) method, call base.ValidateToken, then add an extra validation?
Validation rule:
1. The header must be "Holder-of-key"
2. The claim "x5t#S256" must exist

Reply:
We need a HttpRequestMessage to get the Client certificate